docs/attributes-registry/process.md

```diff
@@ -11,12 +11,32 @@
 | `process.command` | string | The command used to launch the process (i.e. the command name). On Linux based systems, can be set to the zeroth string in `proc/[pid]/cmdline`. On Windows, can be set to the first parameter extracted from `GetCommandLineW`. | `cmd/otelcol` |
 | `process.command_args` | string[] | All the command arguments (including the command/executable itself) as received by the process. On Linux-based systems (and some other Unixoid systems supporting procfs), can be set according to the list of null-delimited strings extracted from `proc/[pid]/cmdline`. For libc-based executables, this would be the full argv vector passed to `main`. | `[cmd/otecol, --config=config.yaml]` |
 | `process.command_line` | string | The full command used to launch the process as a single string representing the full command. On Windows, can be set to the result of `GetCommandLineW`. Do not set this if you have to assemble it just for monitoring; use `process.command_args` instead. | `C:\cmd\otecol --config="my directory\config.yaml"` |
+| `process.end` | string | The date and time the process ended, in ISO 8601 format. | `2023-11-21T09:26:12.315Z` |
```

Comment:
I think this and `process.exit_code` are strange as resource attributes. 

If it's a long running process, there is no end or exit code yet. Once we have them, I guess they could be added as resource attributes, but then that means the resource has tangibly changed even though it's representing the same resource. So I don't see how these are useful as resource attributes, since resources are sort of expected to be immutable.

Reply:
These fields are only in the attributes-registry, and not in the process resource schema, for this reason. You're right that it wouldn't really make sense to have these values in the immutable resource when they probably wouldn't be known at creation time, and couldn't be added later because of resource immutability.

The intention of having these in the attributes-registry is they could be used, for example, as attributes on a "process-exit" event log, where the values are known.

Reply:
@mjwolf We would need to split the yaml files into two groups then (for resource and non-resource).

I think the best way to do so, is to move existing process attributes to registry, first. and then only extend the registry yaml files.

Reply:
The problem here is that the entire group (in yaml) is defined as resource type.
To fix that properly, we need to resolve https://github.com/open-telemetry/build-tools/issues/242 first. 
So, I fear this is another case being blocked on https://github.com/open-telemetry/build-tools/issues/242.

Which however should not block us on discussing the general things here.

Reply:
@braydonk I would also add that these are not resource attributes despite they are defined in the yaml file as such. There were just normal attributes and change to `resource` was a quick fix until a solution for building the tools will be implemented, see https://github.com/open-telemetry/semantic-conventions/pull/524

Reply:
Thanks for the clarification, I didn't know about this issue with the build tools and mistook this to mean they were intended to be resource attributes. Assuming that they are not intended to be resource attributes and would be changed once that issue is resolved, then this is fine with me.

Reply:
https://github.com/open-telemetry/build-tools/issues/242 shouldn't be blocking this any longer